src/main/java/com/odysseusinc/arachne/portal/model/listeners/AnalysisChangesListener.java

```diff
@@ -0,0 +1,24 @@
+package com.odysseusinc.arachne.portal.model.listeners;
+
+import com.odysseusinc.arachne.portal.model.Analysis;
+import com.odysseusinc.arachne.portal.model.AnalysisFile;
+
+import javax.persistence.PrePersist;
+import javax.persistence.PreRemove;
+import javax.persistence.PreUpdate;
+import java.util.Date;
+
+public class AnalysisChangesListener {
+
+    @PreUpdate
+    @PrePersist
+    @PreRemove
+    public void update(Object entity) {
+
+        if (entity instanceof Analysis) {
+            ((Analysis) entity).getStudy().setUpdated(new Date());
+        } else if (entity instanceof AnalysisFile) {
+            ((AnalysisFile) entity).getAnalysis().getStudy().setUpdated(new Date());
+        }
```

Comment:
It's not obvious to me, how modified study is persisted into the database?

Reply:
Not sure what exactly the question
Hibernate does it, see 1.2:
https://howtodoinjava.com/hibernate/hibernate-entity-persistence-lifecycle-states/

A persistent entity has been associated with a database table row and it’s being managed by the current running Persistence Context. Any change made to such entity is going to be detected and propagated to the database during the Session flush-time

Reply:
I meant there's no explicit call of save method for study, how to ensure it persist changes?

Reply:
Study record already exists when we modify analysis, any change to study will be propagated on transaction commit
Also, it is @PrePersist event handler so we know that saving already started for the context